src/sdc11073/pysoap/soapclient.py

```diff
@@ -249,132 +257,100 @@ def post_message_to(self, path: str,
             raise HTTPReturnCodeError(http_response.status, http_response.reason, soap_fault)
         return message_data
 
-    def _send_soap_request(self, path: str, xml: bytes | str, msg: str) -> tuple[HTTPResponse, str]:
-        """Send SOAP request using HTTP."""
-        if not isinstance(xml, bytes):
-            xml = xml.encode('utf-8')
+    def _send_soap_request(self, path: str, xml: bytes, log_msg: str) -> tuple[HTTPResponse, bytes]:
+        """Send SOAP request."""
+        commlog.get_communication_logger().log_soap_request_out(xml, 'POST')
+        self._log.debug("{}:POST to netloc='{}' path='{}'", log_msg, self._netloc, path)
 
         headers = {
             'Content-type': 'application/soap+xml; charset=utf-8',
             'user_agent': 'pysoap',
             'Connection': 'keep-alive',
         }
-        commlog.get_communication_logger().log_soap_request_out(xml, 'POST')
-
+        # set accepted encodings
         if self.supported_encodings:
             headers['Accept-Encoding'] = ','.join(self.supported_encodings)
+        # if possible encode ( compress) xml data
         if self.request_encodings:
             for compr in self.request_encodings:
                 if compr in self.supported_encodings:
                     xml = CompressionHandler.compress_payload(compr, xml)
                     headers['Content-Encoding'] = compr
                     break
+        # split message into chunks?
         if self._chunked_requests:
             headers['transfer-encoding'] = "chunked"
             xml = mk_chunks(xml)
```

Comment:
instead of bool `_chunked_requests` this could be an `int`, where `-1` is no chunked requests and otherwise the chunk size which is passed to `mk_chunks`

Reply:
good idea. Changed all occurances of that flag to a chunk size integer, 0 means no chunks.